calculator.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+from sys import path
+# from typing import Protocol,ContextManager,runtime_checkable
+
+def create_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", default= None ,help = "equation file")
+    args = parser.parse_args()
+    return args
+
+def take_expresion(file_name) -> str:
+    try: 
+        with open(file_name,'r') as f:
+            return f.read()
+
+    except (FileNotFoundError, TypeError) :
+        print('file is not exsist')
+        print("write your expression:")
+        equation = input("> ") 
+        return equation
+
+    
+class Calculator():
+
+    def calculate(self,equation: str) -> float:
+        equation_list = []
+        for c in equation:
+            equation_list.append(c)
+        return self.solver(equation_list)
+    
+    def solver(self, pre_calculate_list: list)-> float:
+
+        if len(pre_calculate_list) == 0:
+            return 0
+
+        stack = []
+        sign = '+'
+        num = 0
+        decimals =0
+
+        while len(pre_calculate_list) > 0:
+            c = pre_calculate_list.pop(0)
+
+            if c.isdigit() and sign != '.':
+                num = num*10+float(c)
+
+            if c.isdigit() and sign == '.':
+                decimals +=1
+                num += (float(c))*(0.10**decimals)
+
+            if len(pre_calculate_list) == 0 or (c == '+' or c == '-' or c == '*' or c == '/' or c == '.'):
+                if sign == '+':
+                    stack.append(num)
+
+                elif sign == '-':
+                    stack.append(-num)
+
+                elif sign == '*':
+                    stack[-1] = self.multiply(stack[-1],num)
+
+                elif sign == '/':
+                    stack[-1] = self.divide(stack[-1],num)
+
+                elif sign == '.':
+                    stack.append(num)
+
+                decimals =0
+                sign = c
+                num = 0
+        return self.sum(stack)
+
+
+    def sum(self,stack: list) -> float:
+        if len(stack)==1:
```

Comment:
This check is redundant. This is what will happen even if you don't check that len==1. len==0 on the other hand will cause an error.